### DISCUSSION TITLE
SMTP settings are not accepting mail servers with self-signed certificates

### DISCUSSION THREAD

**Discussion by setzerbomb** [pocketbase/pocketbase#669] [Q&A]
Hi,

I've been trying to use a mail server configured with a self-signed certificate, but this error appears:

![image](https://user-images.githubusercontent.com/6942774/193115995-c37a6ad6-783d-43f6-ad50-66e23affcce6.png)

Are there any configuration steps that I'm missing?

**Comment by ganigeorgiev** [comment]
The std `crypto/tls` package [has `InsecureSkipVerify` option](https://pkg.go.dev/crypto/tls#Config.InsecureSkipVerify) but as stated in the comment it is really suitable only for tests (or if you have some additional verification).

I've never used self signed certificate for a mail server and I don't know how that really works. We can add support for the `InsecureSkipVerify` configuration, but to be honest for a public facing service this doesn't make much sense to me.

You can try switching to STARTTLS and opt-in for "optimistic" encryption.

_I'll move the issue to Q&A discussion._

**Reply by lewcpe** [reply]
It's useful to have InsecureSkipVerify option for local testing. I test my application within a docker-compose file and use a fake smtp server in a container.

**Comment by gramakri** [comment]
@ganigeorgiev We are trying to package pocketbase for Cloudron. On Cloudron, the mail server is internal. The internal mail server hostname and the (valid Let's Encrypt) certificate don't match. We can submit a PR that makes this setting configurable. Would you be open to adding this option ?

**Reply by ganigeorgiev** [reply]
> Would you be open to adding this option ?

No, I still don't think it will be a good idea. As a workaround you can also load the certificates in the `SSL_CERT_FILE/SSL_CERT_DIR` location (see https://github.com/pocketbase/pocketbase/discussions/4595#discussioncomment-8855032).

**Reply by gramakri** [reply]
Thanks for the suggestion but the issue is not the validity of the cert. The mail server is valid and from Let's Encrypt. The issue is that the internal name (say, 'mail') and the external name (say, 'mail.domain.com') do not match. We have to use internal names so that the name server's public facing DNS can be changed on the fly without interrupting the app.

**Reply by ganigeorgiev** [reply]
@gramakri Then I don't understand your described problem, nor how it is related to the above self-signed certicates discussion.

Feel free to open a new [Q&A post](https://github.com/pocketbase/pocketbase/discussions/new/choose) with more details about your use case and why the current options don't work for you. PRs are welcomed but I don't promise that it will get merged (see also https://github.com/pocketbase/pocketbase/discussions/4355).

**Reply by vladimirdulov** [reply]
> > Would you be open to adding this option ?
> 
> No, I still don't think it will be a good idea. As a workaround you can also load the certificates in the `SSL_CERT_FILE/SSL_CERT_DIR` location (see [#4595 (reply in thread)](https://github.com/pocketbase/pocketbase/discussions/4595#discussioncomment-8855032)).

The relevant error is:
```
Failed to send the test email. Raw error: 
tls: failed to verify certificate: x509: certificate is valid for *.domain.com, not mail.
```

**Reply by philkunz** [reply]
Any resolution in sight for this? We are affected by this as well.

**Reply by ganigeorgiev** [reply]
@philkunz The `failed to verify certificate` error could be returned for a lot of reasons. It is not clear what actually is the problem you are experiencing.
If the comments above or the workaround in https://github.com/pocketbase/pocketbase/discussions/4595#discussioncomment-8854384 doesn't help, please open a new Q&A discussion elaborating more on your setup. But again, there are no plans for now to allow the `InsecureSkipVerify` option.

**Comment by muety** [comment]
I'd actually love to be able to accept self-signed or invalid certificates for testing purposes, especially when using tools like [smtp4dev](https://github.com/rnwood/smtp4dev) locally for debugging.

Specifically, I'm getting the following error when using a self-signed certificate: 

```
tls: failed to verify certificate: x509: certificate relies on legacy Common Name field, use SANs instead.
```

**Reply by muety** [reply]
Any chances to add the option to trust self-signed certificates for testing purposes?
